### ISSUE TITLE
Mime-Type Error office files

### ISSUE THREAD

**Issue by raffaelone** [rotdrop/nextcloud-app-pdf-downloader#28]
I've installed the app and all the required dependencies, but I get always a MIME TYPE error converting the office files.
For .odt files `"application/vnd.oasis.opendocument.text"` and `"application/vnd.openxmlformats-officedocument.wordprocessingml.document"` for .docx., etc.
Could it be due to the Libreoffice configuration or something else that I haven't configured correctly?
Thank you very much in advance

**Comment by rotdrop** [comment]
Thank you for reporting, sorry that I did not address this earlier. I will now try to reproduce this problem on my side ...

**Comment by rotdrop** [comment]
I have now created test folder with two office files .odt and .docx and the thing just seems to work on my side. 

- application/vnd.oasis.opendocument.text
- application/vnd.openxmlformats-officedocument.wordprocessingml.document

The question is then why it does not work. Can please be a bit more specific about the error message, what exactly is reported?

Thanks.

**Comment by raffaelone** [comment]
Hello and thanks for the answer. I attach the file generated with the error.
I replicated your same test: a folder named test that contains two office files (file1.odt and file2.docx).

Thanks

[2023-03-01T13_57_06+01_00--test.pdf](https://github.com/rotdrop/nextcloud-app-pdf-downloader/files/10861305/2023-03-01T13_57_06%2B01_00--test.pdf)

**Comment by rotdrop** [comment]
BTW, "installing all the required dependencies", if you open the administrator settings and look at the status of the configured converters, what does it show? Is it very different from the attached screenshot, and in particular: is unoconv mentioned?

![Screenshot_20230301_150436-cropped](https://user-images.githubusercontent.com/1494769/222163595-3cf76625-80e6-44cd-84df-4cad7f469f9b.png)

**Comment by rotdrop** [comment]
And if unoconv is installed, then on the command-line it should be possible to convert a file like follows:

```
unoconv -f pdf --stdin --stdout -e ExportNotes=false < INFILE > PDF_OUTFILE
```

**Comment by raffaelone** [comment]
unoconv is installed and mentioned in the administrator settings, but not works from command-line. If I try to convert a file this is the output:

```
$ unoconv -vf pdf file1.odt 
/usr/bin/unoconv:19: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  from distutils.version import LooseVersion
Input file: file1.odt
Selected output format: Portable Document Format [.pdf]
Output file: file:///home/test-user/file1.pdf
Traceback (most recent call last):
  File "/usr/bin/unoconv", line 1037, in convert
    document.storeToURL(outputurl, tuple(outputprops) )
uno.com.sun.star.io.IOException: SfxBaseModel::impl_store <file:///home/test-user/file1.pdf> failed: 0x507(Error Area:Io Class:Access Code:7) at ./sfx2/source/doc/sfxbasemodel.cxx:3215 at ./sfx2/source/doc/sfxbasemodel.cxx:1789

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/unoconv", line 1282, in <module>
    main()
  File "/usr/bin/unoconv", line 1198, in main
    convertor.convert(inputfn)
  File "/usr/bin/unoconv", line 1039, in convert
    raise UnoException("Unable to store document to %s (ErrCode %d)\n\nProperties: %s" % (outputurl, e.ErrCode, outputprops), None)
  File "/usr/lib/python3/dist-packages/uno.py", line 507, in _uno_struct__getattr__
    return getattr(self.__dict__["value"], name)
AttributeError: ErrCode
```

**Comment by rotdrop** [comment]
impl_store <file:///home/test-user/file1.pdf> 

Is unoconv running  with the correct user-id? From the error message I would suspect that this is perhaps is a permission problem. What happens if you use stdin and stdout as suggested in my previous comment?

**Comment by raffaelone** [comment]
The cursor blinks and nothing happens

**Comment by rotdrop** [comment]
Did you have a look here?

https://github.com/unoconv/unoconv#installing-unoconv

**Comment by raffaelone** [comment]
Yes, I did! 
The problem is the Ubuntu unoconv package.

However that's how I solved it.

First I've uninstalled the unoconv .deb package:
`apt purge --remove unoconv`

Cloned the unoconv repo:
```
git clone https://github.com/dagwieers/unoconv.git
cp unoconv/unoconv /usr/bin
```
Added a symlink to point the /usr/bin/python installing "python-is-python3":
`apt install python-is-python3`

Now unoconv (Recursive PDF Downloader) works correctly in NC.

Thanks for your support and patience
